src/pygambit/game.pxi

```diff
@@ -1162,14 +1162,14 @@ class Game:
         resolved_infoset = cython.cast(Infoset, self._resolve_infoset(infoset, "append_infoset"))
         resolved_nodes = (
                        [self._resolve_node(n, "append_move") for n in nodes]
-                       if hasattr(nodes, "__iter__")
+                       if (hasattr(nodes, "__iter__") and not isinstance(nodes, str))
                        else [self._resolve_node(nodes, "append_move")]
         )
 
         if any(len(rs.children) > 0 for rs in resolved_nodes):
             raise UndefinedOperationError("append_move(): `node` must be a terminal node")
 
-        if hasattr(nodes, "__iter__") and len(nodes) != len(set(nodes)):
+        if hasattr(nodes, "__iter__") and any(nodes.count(n) > 1 for n in nodes):
             raise ValueError("append_move(): Each `node` must be listed only once.")
 
         if hasattr(nodes, "__iter__") and (len(nodes) == 0):
```

Comment:
Likewise multiple issues:

* The parentheses are redundant;
* However, this check should be on `resolved_nodes` instead of `nodes` in which case checking for whether it's iterable is redundant;
* Then the correct way (per PEP 8) to check whether a list is empty is using `not`, not `len() == 0`.